cl/search/documents.py

```diff
@@ -1070,3 +1081,578 @@ def prepare(self, instance):
+
+    def prepare_proceduralHistory(self, instance):
+        return instance.procedural_history
+
+    def prepare_posture(self, instance):
+        return instance.posture
+
+    def prepare_syllabus(self, instance):
+        return instance.syllabus
+
+    def prepare_sibling_ids(self, instance):
+        return [opinion.pk for opinion in instance.sub_opinions.all()]
+
+    def prepare_panel_ids(self, instance):
+        return [judge.pk for judge in instance.panel.all()]
+
+    def prepare_dateFiled(self, instance):
+        if instance.date_filed is None:
+            return
+
+        if isinstance(instance.date_filed, str):
+            datetime_object = datetime.strptime(
+                instance.date_filed, "%Y-%m-%d"
+            )
+            return midnight_pt(datetime_object)
+
+        return midnight_pt(instance.date_filed)
+
+    def prepare_dateArgued(self, instance):
+        if instance.docket.date_argued is None:
+            return
+
+        if isinstance(instance.docket.date_argued, str):
+            datetime_object = datetime.strptime(
+                instance.docket.date_argued, "%Y-%m-%d"
+            )
+            return midnight_pt(datetime_object)
+
+        return midnight_pt(instance.docket.date_argued)
+
+    def prepare_dateReargued(self, instance):
+        if instance.docket.date_reargued is None:
+            return
+
+        if isinstance(instance.docket.date_reargued, str):
+            datetime_object = datetime.strptime(
+                instance.docket.date_reargued, "%Y-%m-%d"
+            )
+            return midnight_pt(datetime_object)
+
+        return midnight_pt(instance.docket.date_reargued)
+
+    def prepare_dateReargumentDenied(self, instance):
+        if instance.docket.date_reargument_denied is None:
+            return
+
+        if isinstance(instance.docket.date_reargument_denied, str):
+            datetime_object = datetime.strptime(
+                instance.docket.date_reargument_denied, "%Y-%m-%d"
+            )
+            return midnight_pt(datetime_object)
+
+        return midnight_pt(instance.docket.date_reargument_denied)
+
+    def prepare_neutralCite(self, instance):
+        neutral_citations = instance.citations.filter(type=Citation.NEUTRAL)
+        if neutral_citations.count():
+            return str(neutral_citations[0])
+        return ""
+
+    def prepare_lexisCite(self, instance):
+        lexis_citations = instance.citations.filter(type=Citation.LEXIS)
+        if lexis_citations.count():
+            return str(lexis_citations[0])
+        return ""
+
+    def prepare_timestamp(self, instance):
+        return datetime.utcnow()
+
+
+@opinion_index.document
+class OpinionDocument(OpinionBaseDocument):
+    id = fields.IntegerField(
+        attr="pk",
+        fields={
+            "raw": fields.KeywordField(attr="pk"),
+        },
+    )
+    author_id = fields.IntegerField()
+    type = fields.KeywordField(attr="type")
```

Comment:
I noticed in the Solr mapping, this is a TextField that uses `text_en_splitting_cl`. Here, we have this one as a `KeywordField` and `type_text` as a `TextField`. Do we require both?

Reply:
I've added the type_text field to display the “human-readable” value of the field in the frontend, and I included the `type` field to enable its utilization in the API response once we develop that feature.

I considered using [Multi-fields](https://www.elastic.co/guide/en/elasticsearch/reference/current/multi-fields.html), but the values of the fields are distinct.

Reply:
Oh I see, yeah the `type_text` is now used in the template, so it's required.